midend/removeParameters.cpp

```diff
@@ -78,7 +78,7 @@ void FindActionParameters::postorder(const IR::MethodCallExpression* expression)
 
 const IR::Node* DoRemoveActionParameters::postorder(IR::P4Action* action) {
     LOG1("Visiting " << dbp(action));
-    BUG_CHECK(getContext()->node->is<IR::IndexedVector<IR::Declaration>>(),
+    BUG_CHECK(getParent<IR::P4Control>() || getParent<IR::P4Program>(),
```

Comment:
It's just ensuring that the frontend has flagged errors for any actions in unexpected places (in actions or parsers or ???)  If we later add actions in parsers, this bug check will ensure this code is updated to handle it.